### PR TITLE
Enable restoring a cluster after disaster recovery

### PR DESCRIPTION
**Pull request by travisn**
**Description of your changes:**
When testing the scenario of restoring a cluster after rebuilding k8s (described [here](https://github.com/rook/rook/issues/1653#issuecomment-536696481)), there are some small fixes to enable the scenario:
1. When creating mon quorum, immediately create all mon deployments. The operator was waiting for quorum after starting the first one, but at least two mons needed to be up. After starting all mons, then check for quorum.
1. Certain requests to mgr modules are hanging while the OSDs are still down. Changed this to timeout when these requests are not responsive. 
1. Unrelated change to reduce operator verbose logging around checking for clean PGs.

**Which issue is resolved by this Pull Request:**
Related to #1653 

**Checklist:**

- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [ ] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.

### CODE REVIEW

File: pkg/operator/ceph/cluster/mon/mon.go [hunk 1]
```diff
@@ -654,6 +654,20 @@ func (c *Cluster) startDeployments(mons []*monConfig, requireAllInQuorum bool) e
 		return fmt.Errorf("cannot start 0 mons")
 	}
 
+	// If all the mon deployments don't exist, allow the mon deployments to all be started without checking for quorum.
+	// This will be the case where:
+	// 1) New clusters where we are starting one deployment at a time. We only need to check for quorum once when we add a new mon.
+	// 2) Clusters being restored where no mon deployments are running. We need to start all the deployments before checking quorum.
+	onlyCheckQuorumOnce := false
+	deployments, err := c.context.Clientset.AppsV1().Deployments(c.Namespace).List(metav1.ListOptions{LabelSelector: "app=rook-ceph-mon"})
```
Comment (leseb) [inline pkg/operator/ceph/cluster/mon/mon.go]:
```suggestion
	deployments, err := c.context.Clientset.AppsV1().Deployments(c.Namespace).List(metav1.ListOptions{LabelSelector: fmt.Sprintf("app=%s", AppName)})
```

**Comment by travisn** [comment]
> Isn't onlyCheckQuorumOnce the exact opposite of requireAllInQuorum? If so, do we need onlyCheckQuorumOnce, can't we just set requireAllInQuorum: true if len(deployments.Items) == len(mons)? I think that's what you want. Hope I got this right :)

There is a subtle difference between these settings. Even if `requireAllInQuorum` is false, it still requires quorum to exist. In the mon disaster recovery scenario, no mon deployments would exist. So we need to start all the mon deployments before we ever check for mon quorum. Otherwise, we will never get a response from mon quorum because only one mon will be started when we check for quorum.

File: pkg/operator/ceph/cluster/mgr/orchestrator_test.go [hunk 2]
```diff
@@ -44,6 +45,10 @@ func TestOrchestratorModules(t *testing.T) {
 				return "", nil
 			}
 		}
+		return "", fmt.Errorf("unexpected ceph command '%v'", args)
```
Comment (BlaineEXE) [inline pkg/operator/ceph/cluster/mgr/orchestrator_test.go]: I don't see a check for an error-positive state in this test.

Comment (travisn) [reply pkg/operator/ceph/cluster/mgr/orchestrator_test.go]: What check are you referring to? This test just had to be updated to implement the `MockExecuteCommandWithOutputFileTimeout()` method in order to catch the events for the module creation as a result of the timeout calls.

Comment (BlaineEXE) [reply pkg/operator/ceph/cluster/mgr/orchestrator_test.go]: There are no tests that `assert.Error(t, err)`, which means there is no coverage of the error state.

Comment (travisn) [reply pkg/operator/ceph/cluster/mgr/orchestrator_test.go]: some negative tests are added now

**Pull request merged**